src/Access/ExternalAuthenticators.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <Access/LDAPParams.h>
+#include <Core/Types.h>
+
+#include <map>
+#include <memory>
+#include <mutex>
+
+
+namespace Poco
+{
+    class Logger;
+
+    namespace Util
+    {
+        class AbstractConfiguration;
+    }
+}
+
+
+namespace DB
+{
+
+class ExternalAuthenticators
```

Comment:
The name of this class seems to be too general. In fact this class is designed to manage only LDAP params. And if in the future we want to support another SSO system we'd more likely add another class. I suppose to rename this class into something like `LDAPServerParamsManager`.

Reply:
The Idea was that if in future another authentication method is added, its params, or even a shared object that represents connection etc., could be placed here, and `getLDAPServerParams` will be changed to something like `getParams(type, name)`.

Reply:
ok